storage/replica.go

```diff
@@ -193,11 +196,12 @@ var _ client.Sender = &Replica{}
 // NewReplica initializes the replica using the given metadata.
 func NewReplica(desc *roachpb.RangeDescriptor, rm *Store) (*Replica, error) {
 	r := &Replica{
-		store:       rm,
-		cmdQ:        NewCommandQueue(),
-		tsCache:     NewTimestampCache(rm.Clock()),
-		sequence:    NewSequenceCache(desc.RangeID),
-		pendingCmds: map[cmdIDKey]*pendingCmd{},
+		store:          rm,
+		cmdQ:           NewCommandQueue(),
+		tsCache:        NewTimestampCache(rm.Clock()),
+		sequence:       NewSequenceCache(desc.RangeID),
+		pendingCmds:    map[cmdIDKey]*pendingCmd{},
+		raftEntryCache: newRaftEntryCache(defaultRaftEntryCacheSize),
```

Comment:
A cache per `Replica` could get quite expensive, especially if each replica's cache could grow to 800 entries (how did you arrive at this number?). I was thinking of a global (per-store) cache keyed by the (RangeID, index) pair.